src/test/kotlin/com/example/flashcardbackend/deck/DeckControllerTest.kt

```diff
@@ -0,0 +1,186 @@
+package com.example.flashcardbackend.deck
+
+import com.example.flashcardbackend.flashcard.FlashcardListItem
+import org.hamcrest.Matchers
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+
+@WebMvcTest(DeckController::class, DeckService::class)
+class DeckControllerTest(
+    @Autowired
+    private val mockMvc: MockMvc,
+) {
+
+    @MockBean
+    private lateinit var deckRepository: DeckRepository
+
+    @Test
+    fun `should return all decks`() {
+        val deckListItems = listOf(
+            DeckListItem(12, 1, "Deck 1", "Description 1"),
+            DeckListItem(13, 1, "Deck 2", "Description 2"),
+        )
+
+        `when`(deckRepository.findAll()).thenReturn(deckListItems)
+
+        mockMvc.perform(get("/decks"))
```

Comment:
When you split your tests into smaller tests, you end adding duplicate code to your test suite. This code sends HTTP requests to the system under test with Spring MVC Test. You can remove this duplicate code by using request builders: https://www.petrikainulainen.net/programming/spring-framework/spring-from-the-trenches-cleaning-up-our-test-code-with-http-request-builders/

Reply:
Added request builder when refactor tests into nested classes :)